tests/test_endpoint_buffers.py

```diff
@@ -38,7 +39,172 @@ def forward_setup_fixture():
+            self, get_json_dict_memory_allocation_size_mock, general_setup, caplog):
+        """The preparation of results fails if the output is too big."""
+        area, _ = general_setup
+        endpoint_buffer = SimulationEndpointBuffer(
+            job_id="JOB_1",
+            random_seed=41,
+            area=area,
+            should_export_plots=False)
+
+        get_json_dict_memory_allocation_size_mock.return_value = 128000
+        with caplog.at_level(logging.WARNING):
+            output = endpoint_buffer.prepare_results_for_publish()
+            assert "Do not publish message bigger than 64 MB, current message size 128.0 MB." \
+                in caplog.text
+
+        assert output == {}
+
+    def test_generate_json_report(self, general_setup):
+        area, _ = general_setup
+        endpoint_buffer = SimulationEndpointBuffer(
+            job_id="JOB_1",
+            random_seed=41,
+            area=area,
+            should_export_plots=False)
+
+        # The ResultsHandler class should be tested as a different unit, so we just mock its output
+        results_handler_mock = MagicMock(all_raw_results={"mocked-results": "some-results"})
+        endpoint_buffer.results_handler = results_handler_mock
+
+        assert endpoint_buffer.generate_json_report() == {
+            "job_id": "JOB_1",
+            "random_seed": 41,
+            "status": "",
+            "progress_info": {
+                "eta_seconds": 0,
+                "elapsed_time_seconds": 0,
+                "percentage_completed": 0
+            },
+            "simulation_state": {"general": {}, "areas": {}},
+            "mocked-results": "some-results"
+        }
+
+    def test_general_results_are_generated(  # pylint: disable-msg=too-many-locals
+            self, general_setup):
+        """Test results are being correctly generated with respect to
+        spot market"""
+        area, slot_length = general_setup
+
+        progress_info = MagicMock()
+        endpoint_buffer = SimulationEndpointBuffer("JOB_1", 41, area, False)
+
+        # add bids/offers/trades
+        start_time = DateTime(2020, 1, 1, 0, 15)
+        for market in area.spot_market():
+            current_time = start_time
+
+            market.bids = {}
+            market.offers = {}
+            market.trades = []
+
+            for i in range(2):
+                market.bids[uuid4()] = self._generate_order(
+                    creation_time=current_time - slot_length,
+                    time_slot=f"TIME_SLOT_{i}")
+
+                market.offers[uuid4()] = self._generate_order(
+                    creation_time=current_time - slot_length,
+                    time_slot=f"TIME_SLOT_{i}")
+
+                market.trades.extend([
+                    self._generate_order(
+                        creation_time=current_time - slot_length,
+                        time_slot=f"TIME_SLOT_{i}") for i in range(2)])
+            current_time += slot_length
```

Comment:
Why you are adding to the `current_time` here, but overwrite this variable in line 187?